formats/qcow-efi.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, modulesPath, ... }:
+{
+  # for virtio kernel drivers
+  imports = [
+    "${toString modulesPath}/profiles/qemu-guest.nix"
+  ];
+
+  options = {
+    diskSize = lib.mkOption {
+      default = "auto";
+      description = "The disk size in megabytes of the system disk image.";
+      type = with lib.types; oneOf [ ints.positive (enum [ "auto" ])];
+    };
+  };
+
+  config = {
+    fileSystems."/" = {
+      device = "/dev/disk/by-label/nixos";
+      autoResize = true;
+      fsType = "ext4";
+    };
+
+    fileSystems."/boot" = {
+      device = "/dev/disk/by-label/ESP";
+      fsType = "vfat";
+    };
+
+    boot.growPartition = true;
+    boot.kernelParams = [ "console=ttyS0" ];
+    boot.loader.grub.device = "nodev";
+
+    boot.loader.grub.efiSupport = true;
+    boot.loader.grub.efiInstallAsRemovable = true;
+    boot.loader.timeout = 0;
+
+    system.build.qcow = import "${toString modulesPath}/../lib/make-disk-image.nix" {
+      inherit lib config pkgs;
+      diskSize = config.diskSize;
```

Comment:
If we go with the small boot size than the `bootSize` parameter could be reduced a bit.

Reply:
Addressed in [062cb2f](https://github.com/nix-community/nixos-generators/pull/204/commits/062cb2f55564ff53b3381d7a07369f469403df6e)